chia/data_layer/s3_plugin_service.py

```diff
@@ -160,20 +163,65 @@ async def download(self, request: web.Request) -> web.Response:
             if not should_download:
                 return web.json_response({"downloaded": False})
 
-            bucket = parse_result.netloc
+            bucket_str = parse_result.netloc
+            my_bucket = self.boto_resource.Bucket(bucket_str)
             target_filename = self.server_files_path.joinpath(filename)
             # Create folder for parent directory
             target_filename.parent.mkdir(parents=True, exist_ok=True)
             log.info(f"downloading {url} to {target_filename}...")
             with concurrent.futures.ThreadPoolExecutor() as pool:
                 await asyncio.get_running_loop().run_in_executor(
-                    pool, functools.partial(self.boto_client.download_file, bucket, filename, str(target_filename))
+                    pool, functools.partial(my_bucket.download_file, filename, str(target_filename))
                 )
         except Exception as e:
             log.error(f"failed parsing request {request} {type(e).__name__} {e}")
             return web.json_response({"downloaded": False})
         return web.json_response({"downloaded": True})
 
+    async def add_missing_files(self, request: web.Request) -> web.Response:
+        try:
+            data = await request.json()
+            store_id = bytes32.from_hexstr(data["store_id"])
+            bucket_str = self.get_bucket(store_id)
+            files = json.loads(data["files"])
+            my_bucket = self.boto_resource.Bucket(bucket_str)
+            existing_file_list = []
+            for my_bucket_object in my_bucket.objects.all():
+                existing_file_list.append(my_bucket_object.key)
```

Comment:
I personally like the comprehension better, but using a set is the functional change.

```suggestion
            existing_files = {my_bucket_object.key for my_bucket_object in my_bucket.objects.all()}
```

Reply:
i think its less readable